src/engine/Server.cpp

```diff
@@ -630,7 +650,19 @@ boost::asio::awaitable<void> Server::processQuery(
     queryExecutionTree = qp.createExecutionTree(pq);
     auto& qet = queryExecutionTree.value();
     qet.isRoot() = true;  // allow pinning of the final result
-    qet.recursivelySetTimeoutTimer(timeoutTimer);
+    // TODO<RobinTF> register abortion handle somewhere
+    auto abortionHandle = std::make_shared<ad_utility::AbortionHandle>();
+    std::optional<
+        ad_utility::unique_cleanup::UniqueCleanup<std::function<void()>>>
+        cancelTimeout = std::nullopt;
+    if (timeLimit.has_value()) {
+      qet.getRootOperation()->recursivelySetTimeConstraint(timeLimit.value());
+      cancelTimeout.emplace(
+          co_await cancelAfterDeadline(abortionHandle, timeLimit.value()),
+          [](auto&& func) { std::invoke(AD_FWD(func)); });
+    }
+    qet.getRootOperation()->recursivelySetAbortionHandle(
+        std::move(abortionHandle));
```

Comment:
Feel free to propose a simpler solution. The main reason I can't use `OnDestructionDontThrow` or `absl::Cleanup` is because the latter requires a deduced type and the former is not move assignnable or something like that. I don't remember the exact details from 2 weeks ago, but I wasn't able to use a simpler type for this task